servlet/src/test/java/io/undertow/servlet/test/dispatcher/DispatcherForwardTestCase.java

```diff
@@ -255,4 +255,21 @@ public void testIncludesPathParameters() throws IOException {
             client.getConnectionManager().shutdown();
         }
     }
+
+    @Test
+    public void testIncludesUrlInPathParameters() throws IOException {
+        TestHttpClient client = new TestHttpClient();
+        try {
+            HttpGet get = new HttpGet(DefaultServer.getDefaultServerURL() + "/servletContext/dispatch?url=http://test.com");
+            get.setHeader("forward", "/path;pathparam=foo");
```

Comment:
I think that the query parameter with the slash should go to the header, not the initial url. The current test would work in the same way without your fix. Something like this:

```java
HttpGet get = new HttpGet(DefaultServer.getDefaultServerURL() + "/servletContext/dispatch?a=b");
 get.setHeader("forward", "/path;pathparam=foo?url=http://test.com");
```

Reply:
Yep! Good catch!
Changes pushed. ✅ 
Verified the test failed before the fix
<img width="880" alt="Before fix" src="https://user-images.githubusercontent.com/25759492/192858529-2aa26c88-0898-46a8-a1a1-6db39916b3e7.png">

And the test passed after the fix!

Reply:
Thanks a lot! It's OK now. Just a nitty request... Do you mind to squash the two commits into a single one?

Reply:
Done! ✅ 
Wondering if branch 2.2.X still accept changes, if so, I can graft this fix to 2.2.X branch and create a new PR. Thanks!

Reply:
@YaoFu-ea thank you! There is no need, the bug fixes are backported weekly or bi-weekly to 2.2.x

Reply:
That's great! 👏 Thank you for the info!